tox.ini

```diff
@@ -10,6 +10,7 @@ skip_missing_interpreters = True
 [testenv]
 basepython = python3
 setenv = PYTHONPATH={toxinidir}
+allowlist_externals = pytest
```

Comment:
Why this is needed? We are adding functional tests, which are using zaza and not pytest.

Reply:
Because zaza uses pytest and if we do not define in tox.ini gives error

Reply:
Where, we never used this with any func tests and we used pytest a lot a never define it in `allowlist_externals`. So I'm confused here.

Reply:
sorry it is a tox4 issue. Since we are using tox3 I removed it